models/transmission_record_test.go

```diff
@@ -25,37 +24,6 @@ var TestSentTransRecord = TransmissionRecord{Response: "ok", Sent: 123, Status:
 var TestEmptyRespTransRecord = TransmissionRecord{Sent: 123, Status: TransmissionStatus(Sent)}
 var TestEmptyTransRecord = TransmissionRecord{}
 
-func TestTransmissionRecord_MarshalJSON(t *testing.T) {
```

Comment:
Why remove the tests?  Are they not still valid?

Only calling out this instance; there may be others.

Reply:
The method no longer exists and is functionally duplicated by `_String()` anyway.

It does not need a test if it does not exist. We do not need to test the core library.

Reply:
Yes, and...  ...it begets the question of why test the MarshalJSON methods directly at all if we've got a separate method that marshals the structure and returns JSON that has its own tests.

MarshalJSON -- while part of the receiver's public interface -- is magical and isn't intended to be called directly.

Reply:
> the question of why test the MarshalJSON methods directly at all

I absolutely agree. This issue's scope is constantly creeping so I don't want to delete all the MarshalJSON tests, but they are pretty useless on the whole.

> MarshalJSON -- while part of the receiver's public interface -- is magical and isn't intended to be called directly.

Again, `transmission` doesn't have a MarshalJSON method. I absolutely refuse to have a test for a method we don't have. Should we also have tests for length methods? Typeofs?

Reply:
Sorry, I was confused.

`transmission_record` no longer has a custom marshaller. Since it does not have any custom logic, it does not need any tests. We cannot test a method if it doesn't exist.

The MarshalJSON methods on the whole are pretty useless, but I think deleting all of them is a bad idea when they aren't hurting anything.

Reply:
I think I'm not communicating well.

I meant to say that if we've got tests around MarshalJSON methods and tests around String() methods (whose implementation calls json.Marshal which subsequently magically calls the aforementioned MarshalJSON method), it seems like we've got duplication in the tests and the MarshalJSON tests should be removed (regardless of whether or not a MarshalJSON method exists for the receiver being tested).

As for deleting MarshalJSON methods, we agreed based on Trevor's input to remove those that aren't doing custom marshalling (i.e. have no reason to exist).  I'm not sure if you're questioning the wisdom of that or ...?  

I know there's a sense of scope creep here.  I'd prefer to frame it as following [the Boy Scout rule](https://medium.com/@biratkirat/step-8-the-boy-scout-rule-robert-c-martin-uncle-bob-9ac839778385).  The existing model code reeks in places and it makes sense -- given that we're actively modifying it to address this issue -- to spend a little extra energy to clean it up a bit.